packages/microcosm/src/subject.js

```diff
@@ -1,140 +1,119 @@
-// @flow
+/**
+ * @fileoverview A subject is a type of observable that is useful when:
+ *
+ * 1. You need a value right away.
+ * 2. A job should run only once, but be subscribed to many times.
+ *
+ * Unlike Observables, when you subscribe to a Subject you immediately
+ * receive a "next" callback with the current payload. This is important
+ * for a couple of use cases that require immediate access to state:
+ *
+ * 1. Domains
+ * 2. Actions
+ * 3. Presenter view models
+ *
+ * @flow
+ */
+import assert from 'assert'
 import { type Microcosm } from './microcosm'
-import { Observable, Observer, getObservable } from './observable'
+import {
+  Observable,
+  type SubscriptionObserver,
+  type Cleanup
+} from './observable'
 import { observable } from './symbols'
-import { noop, EMPTY_SUBSCRIPTION } from './empty'
 import { set, merge } from './data'
+import { isPromise, isObservable, isObject, isPlainObject } from './type-checks'
 
-export class Subject {
+export class Subject extends Observable {
```

Comment:
A really nice side-effect of extending from Observable is that the scheduler and helper methods like `.map` and `.subscribe` can be shared. It truly is an extension of Observable now.